tools/redcap2linst.php

```diff
@@ -36,17 +38,37 @@
 
 $headers     = fgetcsv($fp);
 $instruments = [];
+$badMap = 0;
+$mapped = 0;
 while ($row = fgetcsv($fp)) {
     $inst = $row[1];
     if (!isset($instruments[$inst])) {
         $instruments[$inst] = [];
     }
-    $linstFormat = toLINST($row[3], $row[0], $row[4], $row[5]);
+    $fieldname = $row[0];
+    if ($trimName) {
+        $formname = $row[1];
+        if (strpos($fieldname, $formname) !== 0) {
+            $badMap++;
+            fwrite(STDERR, "Field $fieldname does not have form name $formname as a prefix\n");
```

Comment:
@driusan i feel like this is very project specific. it's not a LORIS requirement to prefix instrument name to the field name, this won't work for maybe projects outside of HBCD i would guess

Reply:
it's specific to the `--trim-formname` option?

LORIS prefixes the instrument name to the field name to make it unique, while redcap has a requirement that fieldnames be unique across all instruments, so if it's not  trimmed it gets duplicated in LORIS, which is why the `--trim-formname` option exists. 

It does apply to HBCD but I would expect it to be the most common naming convention used to meet redcap's globally unique fieldname requirement and this code only applies if the option for it is explicitly passed.

Reply:
i see, sorry i misunderstood, i thought it was doing the opposite, got it!